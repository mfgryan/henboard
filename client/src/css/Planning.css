.PlanningTron{
    border: 2px solid #DDD;
}
.PlanningTextArea{
    width: 100%;
    min-height: 200px;
    max-width: 100%;
    margin: 20px 0 5px 0;
}
.PlanningButton {
    text-align: right;
}
