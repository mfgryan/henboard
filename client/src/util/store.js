// redux dep
import { createStore, applyMiddleware } from "redux";
import { logger, validate, fn } from "../util/middleware.js"; // eslint-disable-line no-unused-vars
import rootReducer from "../reducers/reducers.js";

export default (function(){
    let store;
    return !store ? createStore(rootReducer, applyMiddleware(validate, logger, fn)) : store;
}());
