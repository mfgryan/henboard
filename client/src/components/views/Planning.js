// react dep
import React, { Component } from "react";
import { connect } from "react-redux";
import { Row, Col } from "react-bootstrap";
import projects from "../../models/projects.js";
import sprints from "../../models/sprints.js";

// component dep
import App from "../App";

// style dep
import "../../css/Planning.css";

export const Planning = ({ project, week }) => {
    return (
        <App>
            <Row>
                <Col md={8} mdOffset={2}>
                    <h2>Mission Statement</h2>
                    <textarea/>
                </Col>
            </Row>
        </App>
    );
};

const mapStateToProps = (state, ownProps) => {
    let project = projects.getCurrentProject(state);
    let sprint = sprints.getCurrentSprint(state, project.project);
    return {
        project: project.project,
        week: sprint.week
    }
};

const mapDispatchToProps = (dispatch, ownProps) => {
    return {};
};

export default connect(mapStateToProps,mapDispatchToProps)(Planning);
