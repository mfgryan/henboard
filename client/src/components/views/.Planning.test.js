import React from "react";
import Planning from "./Planning";
import renderer from "react-test-renderer";

// redux dep
import { Provider } from "react-redux";
import { store } from "../../util/store.js";

// router dep
import { BrowserRouter as Router } from "react-router-dom";

it("Planning renders correctly", () => {
    let item = {
        project: "henboard",
        missionStatement: "foo",
        editing: false,
        value: "bar"
    };
    const tree = renderer
        .create(
            <Router>
                <Provider store={store}>
                    <Planning
                        project="henboard"
                        item={item}
                        addStatement={() => {}}
                        changeValue={() => {}}
                        toggleAddStatement={() => {}}
                    />
                </Provider>
            </Router>
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
