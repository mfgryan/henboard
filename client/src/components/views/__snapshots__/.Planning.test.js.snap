exports[`test Planning renders correctly 1`] = `
<div
  className="PlanningTron jumbotron">
  <div
    className="row">
    <div
      className="col-md-8 col-md-offset-2">
      <h2
        onClick={[Function]}>
        Mission Statement
      </h2>
    </div>
  </div>
  <div
    className="row">
    <div
      className="col-md-8 col-md-offset-2">
      <h4>
        Click \"Mission Statement\" to create a new plan
      </h4>
    </div>
  </div>
</div>
`;
